test/lib/ansible_test/_internal/docker_util.py

```diff
@@ -195,10 +195,19 @@ def docker_images(args, image):
     except SubprocessError as ex:
         if 'no such image' in ex.stderr:
             stdout = ''  # podman does not handle this gracefully, exits 125
+        elif 'function "json" not defined' in ex.stderr:
+            # podman > 2 && < 2.2.0 breaks with --format {{json .}}, and requires --format json
+            # So we try this as a fallback. If it fails again, we just raise the exception and bail.
+            stdout, _dummy = docker_command(args, ['images', image, '--format', 'json'], capture=True, always=True)
         else:
             raise ex
-    results = [json.loads(line) for line in stdout.splitlines()]
-    return results
+
+    if stdout[0] == '[':
```

Comment:
Alternatively to https://github.com/ansible/ansible/pull/72096/files#r499654149, you could use a slice operation in order to prevent `IndexError` from being raised when accessing the first element of an empty string:
```suggestion
    if stdout[:1] == '[':
```

Reply:
_(This is still a good suggestion in case if the `docker_command()` call returns an empty stdout)_

Reply:
I think there's no point in doing the comparisons if the image doesn't exist in the first place, so went with your first suggestion of returning early. Good catch. :)

Reply:
@relrod But that's not the only case when this may happen. There's two `docker_command()` invocations. If any of them returns an empty string w/o raising exceptions, it'll cause the same corner case here too.

Reply:
Fixed, by switching to startswith() like you originally suggested.